addons/mail/static/src/models/message_action_list.js

```diff
@@ -1,177 +1,99 @@
-         */
-        onClickReplyTo(ev) {
-            markEventHandled(ev, 'MessageActionList.replyTo');
-            this.messageView.replyTo();
-        },
-        /**
-         * @private
-         * @param {MouseEvent} ev
-         */
-        onClickToggleCompact(ev) {
-            this.update({ isCompact: !this.isCompact });
-        },
-        /**
-         * @private
-         * @param {MouseEvent} ev
-         */
-        onClickToggleStar(ev) {
-            this.message.toggleStar();
-        },
-        /**
-         * @private
-         * @returns {integer|FieldCommand}
-         */
-        _computeActionsCount() {
-            if (this.message){
-                const actions = [
-                    this.hasMarkAsReadIcon,
-                    this.hasReplyIcon,
-                    this.message.canBeDeleted,
-                    this.message.canBeDeleted,
-                    this.message.canStarBeToggled,
-                    this.message.hasReactionIcon,
-                ]
-                return actions.filter(Boolean).length;
-            }
-            return clear();
-        },
-        /**
-         * @private
-         * @returns {string}
+         * @returns {boolean}
          */
-        _computeAddReactionText() {
-            return this.env._t("Add a Reaction");
+        _computeHasCompactAction() {
+            return this.messageView.isInChatWindow && (this.actionsWithoutCompactCount > this.compactThreshold);
         },
         /**
          * @private
          * @returns {boolean}
          */
-        _computeHasCompactIcon() {
-            const COMPACT_THRESHOLD = 2;
-            const viewer = this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.threadViewer;
-            if (viewer && viewer.chatWindow) {
-                return this.actionsCount > COMPACT_THRESHOLD;
+        _computeHasReplyToAction() {
+            if (
+                this.messaging && this.messaging.inbox &&
+                this.message && !this.message.isTemporary && !this.message.isTransient &&
+                this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread && (
+                    this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread === this.messaging.inbox ||
+                    this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread.model === 'mail.channel'
+                )
+            ) {
+                return true;
             }
+            return false;
         },
         /**
          * @private
-         * @returns {boolean}
+         * @returns {FieldCommand}
          */
-        _computeHasMarkAsReadIcon() {
-            return Boolean(
+        _computeMessageActionViews() {
```

Comment:
I don't think we should compute the list here.

Each action view should be computed individually immediately, instead of having "has..." computes.

And then the list should just be the inverse of all existing actions, which removes the need to repeat all the conditions.